infrastructure/running-a-node/setup-full-node.md

```diff
@@ -1,4 +1,258 @@
 ---
 title: Set Up a Full Node
-description: TODO
----
\ No newline at end of file
+description: Learn how to set up and secure a full Polkadot node. Follow step-by-step instructions for installation, node types, RPC setup, and WebSocket security.
+---
+
+## Introduction
+
+If you're building dApps or products on a Polkadot SDK-based chain, running a node as a back-end provides increased operational control and decentralization. All Polkadot SDK node implementations include the RPC server, crucial for accessing and interacting with the Polkadot network.
+
+This guide demonstrates connecting to [Polkadot](https://polkadot.network/){target=\_blank}, but the same process applies to any other [Polkadot SDK](https://paritytech.github.io/polkadot-sdk/master/polkadot_sdk_docs/polkadot_sdk/index.html){target=\_blank}-based chain. This tutorial will guide you through configuring, securing, and maintaining a full node. You'll learn about the differences between archive and pruned nodes, how to secure the WebSocket connection, and the steps necessary to safely expose and maintain your node's RPC server for external access.
+
+## Types of Nodes
+
+- **Pruned node** - prunes historical states of all finalized block states older than a specified number except for the genesis block's state
+- **Archive node** - keeps all the past blocks and their states, making it convenient to query the past state of the chain at any given time. Archive nodes use a lot of disk space, which means they should be limited to use cases that require easy access to past on-chain data
+- **Light node** - has only the runtime and the current state but doesn't store past blocks, making them useful for resource-restricted devices
+
+!!!tip
+    On [Stakeworld](https://stakeworld.io/docs/dbsize){target=\_blank}, you can find a list of the database sizes of Polkadot and Kusama nodes.
+
+### State vs. Block Pruning
+
+A pruned node only keeps a limited number of finalized blocks of the network, not its complete history. State and block pruning are two ways of removing old blocks from a system. State pruning removes the states of old blocks while preserving block headers. Block pruning removes the block bodies of old blocks while retaining block headers. You can complete many frequently required actions with a pruned node, such as displaying account balances, making transfers, setting up session keys, and staking. 
+
+## Setup Instructions
```

Comment:
Some of this content needs to be revisited, as it includes instructions for installing the polkadot binary and that should be a separate guide